### PR TITLE
dogehouse: Add deafen

### PR DESCRIPTION
**Pull request by amitojsingh366**
@ityonemo helped in fixing bug, good guy!

### CODE REVIEW

**Comment by amitojsingh366** [comment]
~~wip~~ ok to merge now

**Comment by amitojsingh366** [comment]
bro wtf are these conflicts 😢
image

**Comment by dark-elixir-shibe** [comment]
Not sure that "deafen" is a particularly evocative command - probably "silence" and "unsilence" are better choices (it's confusing as it is, because mute can mean 'speaker' OR 'microphone'.  Also, is there a reason why this change needs to propagate to the BE?  It seems like it should live in the browser, or am I missing something?

**Comment by amitojsingh366** [comment]
> Not sure that "deafen" is a particularly evocative command - probably "silence" and "unsilence" are better choices (it's confusing as it is, because mute can mean 'speaker' OR 'microphone'. Also, is there a reason why this change needs to propagate to the BE? It seems like it should live in the browser, or am I missing something?

I set the command name as "deafen" as that's what discord does and on the front end a few components like user avatars had the "deafened" property on them *(commit history said @benawad did that)* so i assumed that's what we will go with.
I didn't really understand what you meant by this *"propagate to the BE"* . If you meant why this has to be propagated to the API then the answer is that on whoever is deafened's avatar it shows the deafened icon just like mute so that it makes sense to others why someone isn't replying.

**Comment by amitojsingh366** [comment]
oh no new conflicts...

**Comment by amitojsingh366** [comment]
gonna update this to match the new spec

File: kibbeh/src/modules/settings/SoundEffectSettingsPage.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ export const SoundEffectSettings: React.FC<ChatSettingsProps> = () => {
 
         {Object.keys(soundEffectSettings).map((k) => {
           return (
-            <div className={`flex mb-4 items-center`} key={k}>
+            <div className={`flex flex mb-4 items-center`} key={k}>
```
Comment (ajmnz) [inline kibbeh/src/modules/settings/SoundEffectSettingsPage.tsx]: I don't understand why you change things like this that are pointless and aren't related to the pr

Comment (amitojsingh366) [reply kibbeh/src/modules/settings/SoundEffectSettingsPage.tsx]: @ajmnz uhh vscode tells me 2 flex are redundant and makes the file yellow, so i did that

Comment (amitojsingh366) [reply kibbeh/src/modules/settings/SoundEffectSettingsPage.tsx]: wait did i add it or remove it?

Comment (amitojsingh366) [reply kibbeh/src/modules/settings/SoundEffectSettingsPage.tsx]: i was supposed to remove it, but it says i added it. oh i see what happened, i had copied the whole kibbeh folder from my prev PR and pasted it here, git now thinks i added that

Comment (amitojsingh366) [reply kibbeh/src/modules/settings/SoundEffectSettingsPage.tsx]: there are so many redundant flex that i had removed but those are lost somewhere

Comment (amitojsingh366) [reply kibbeh/src/modules/settings/SoundEffectSettingsPage.tsx]: ive added it back now

**Comment by benawad** [comment]
see https://github.com/benawad/dogehouse/blob/staging/kousa/ADDING_TO_MESSAGING_API.md for instructions for new spec

**Comment by amitojsingh366** [comment]
> 
> 
> see https://github.com/benawad/dogehouse/blob/staging/kousa/ADDING_TO_MESSAGING_API.md for instructions for new spec

i don't see what i haven't followed from here? could you please tell me *(also i just copied the exact way mute works...)*

File: kibbeh/tailwind.config.js [hunk 2]
```diff
@@ -55,7 +55,6 @@ module.exports = {
     spacing: {
       0: "0px",
       1: "5px",
```
Comment (benawad) [inline kibbeh/tailwind.config.js]: why are you changing tailwind config?

File: kibbeh/src/lib/showErrorToast.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { useErrorToastStore } from "../modules/errors/useErrorToastStore";
+import { ToastDurations } from "../ui/Toast";
 
-export const showErrorToast = (m: string) => {
+export const showErrorToast = (m: string, duration?: ToastDurations, button?: JSX.Element, onClose?: () => void) => {
   console.log("showErrorToast: ", m);
-  useErrorToastStore.getState().showToast({ message: m });
+  useErrorToastStore.getState().showErrorToast({ message: m, duration, button, onClose });
```
Comment (benawad) [inline kibbeh/src/lib/showErrorToast.ts]: why are you changing toasts for deafen?

**Review by benawad — CHANGES_REQUESTED** [review]
the deafen part looks good, but there is other crap mixed in

File: kibbeh/tailwind.config.js (hunk 2 above)
Comment (amitojsingh366) [reply kibbeh/tailwind.config.js]: i didn't change this, im not sure how it happened. will revert it

File: kibbeh/src/lib/showErrorToast.ts (hunk 3 above)
Comment (amitojsingh366) [reply kibbeh/src/lib/showErrorToast.ts]: don't know why this happened, reverted it

**Pull request merged**